### ISSUE TITLE
Destruction: o3de fails to build Editor.exe and other executables when Blast gem is enabled

### ISSUE THREAD

**Issue by LB-MateuszSzweda** [o3de/o3de#15445]
**Describe the bug**
Building latest Development and Stabilization branch builds with **Blast** gem enabled results in lack of built Editor.exe and other crucial executables such as AssetProcessor.exe, which blocks the Editor from launching with **Blast** gem.

Following errors regarding Blast were observed in the building log:

```
C:\o3de-extras\Gems\Blast\Code\Source\Family/ActorRenderManager.h(51,21): error C2039: 'MaterialAssignmentMap': is not a member of 'AZ::Render' [C:\o3de\build\windows_vs2019\External\Blast-72dcfbc4\Code\Blast.Static.vcxproj]
          AZ::Render::MaterialAssignmentMap m_materialMap;
                      ^
  C:\o3de\Gems\Atom\Feature\Common\Code\Include\Atom/Feature/Mesh/MeshFeatureProcessorInterface.h(24): note: see declaration of 'AZ::Render'
      namespace Render
C:\o3de-extras\Gems\Blast\Code\Source\Family/ActorRenderManager.h(51,43): error C3646: 'm_materialMap': unknown override specifier [C:\o3de\build\windows_vs2019\External\Blast-72dcfbc4\Code\Blast.Static.vcxproj]
          AZ::Render::MaterialAssignmentMap m_materialMap;
                                            ^
C:\o3de-extras\Gems\Blast\Code\Source\Family/ActorRenderManager.h(51,56): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\o3de\build\windows_vs2019\External\Blast-72dcfbc4\Code\Blast.Static.vcxproj]
          AZ::Render::MaterialAssignmentMap m_materialMap;
                                                         ^
C:/o3de-extras/Gems/Blast/Code/Source/Family/ActorRenderManager.cpp(44,137): error C2664: 'AZ::Render::MeshFeatureProcessorInterface::MeshHandle AZ::Render::MeshFeatureProcessorInterface::AcquireMesh(const AZ::Render::MeshHandleDescriptor &,const AZ::Render::CustomMaterialMap &)': cannot convert argument 2 from 'AZ::Render::MaterialAssignmentMap' to 'const AZ::Render::CustomMaterialMap &' [C:\o3de\build\windows_vs2019\External\Blast-72dcfbc4\Code\Blast.Static.vcxproj]
                  m_meshFeatureProcessor->AcquireMesh(AZ::Render::MeshHandleDescriptor{ m_meshData->GetMeshAsset(chunkId) }, m_materialMap);
                                                                                                                                          ^
  C:/o3de-extras/Gems/Blast/Code/Source/Family/ActorRenderManager.cpp(44,137): note: Reason: cannot convert from 'AZ::Render::MaterialAssignmentMap' to 'const AZ::Render::CustomMaterialMap'
                  m_meshFeatureProcessor->AcquireMesh(AZ::Render::MeshHandleDescriptor{ m_meshData->GetMeshAsset(chunkId) }, m_materialMap);
                                                                                                                                          ^
  C:/o3de-extras/Gems/Blast/Code/Source/Family/ActorRenderManager.cpp(44,124): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
                  m_meshFeatureProcessor->AcquireMesh(AZ::Render::MeshHandleDescriptor{ m_meshData->GetMeshAsset(chunkId) }, m_materialMap);
```

No workaround was found as building with target **ALL_BUILD**, building other projects than **AutomatedTesting**, using **Visual Studio 2022** and omitting _-DAZ_USE_PHYSX5=OFF_ argument during configuration also resulted in the issue above.

Please note that building Editor first, enabling **Blast** gem and rebuilding the Editor also resulted in these errors present in the log and **Blast** components not appearing in Entity Outliner.

For further information please refer to the attached build log.

**Prerequisites**

* **Blast**  gem was obtained from cloned https://github.com/o3de/o3de-extras/tree/blast-experimental repository.
* Destruction assets from cloned _o3de-extras\Gems\Blast\AutomatedTesting\Assets\Destruction_ directory were copied into
an active o3de project directory.

**Steps to reproduce**

1. Obtain Python for cloned o3de build by using _get_python.bat_ from _\o3de\python_ repository_.
2. Register the obtained **Blast** gem by using  _o3de\scripts\o3de.bat register -gp <Path to Blast Gem>_ command.
3. Enable the **Blast** gem for project which will be build by using _o3de\scripts\o3de.bat enable-gem --gem-name Blast --project-name <your project name>_ command.
4. Configure the build with _cmake -B build/windows_vs2019 -G "Visual Studio 16" -DLY_3RDPARTY_PATH=C:\o3de-packages -DLY_PROJECTS=<your project name> -DAZ_USE_PHYSX5=OFF_ command.
5. Build with _cmake --build build/windows_vs2019 --target Editor --config profile -- -m_ command.
6. After building finishes navigate to built  _\build\windows_vs2019\bin\profile_ directory and try to locate the Editor.exe

**Expected behavior**
Editor can be launched with Editor.exe after building with **Blast** gem enabled.

**Actual behavior**
Editor.exe and other executables are not generated after building with **Blast** gem enabled and Editor can not be launched.

**Found in Branch**
Development 
Stabilization/2305

**Commit ID from** [o3de/o3de](https://github.com/o3de/o3de) **Repository**
9f4341f
26273c2

**Desktop/Device:**
- Device: PC
- OS: Windows
- Version: 10 Pro
- CPU: AMD Ryzen 9 3900X
- GPU: NVIDIA GeForce RTX 2060 SUPER
- Memory: 16GB

**Additional context**
[buildlog.txt](https://github.com/o3de/o3de/files/11100427/buildlog.txt)

**Comment by moraaar** [comment]
Fix has been merged to [o3de:blast-experimental](https://github.com/o3de/o3de-extras/tree/blast-experimental) branch
Closing Issue. PR: https://github.com/o3de/o3de-extras/pull/236

**Comment by LB-MateuszSzweda** [comment]
Issue verified as fixed in Development (0c6724e) and Stabilization/2305 (26273c2). No errors were seen in building log after building source configuration using repro steps and the Blast components were usable in Editor.
